### ISSUE TITLE
Visual Studio ignores inlineSourceMaps / inlineSources option of tsconfig.json

### ISSUE THREAD

**Issue by johnnyreilly** [Microsoft/TypeScript#6317]
Hi,

There's mention of this problem in: https://github.com/Microsoft/TypeScript/issues/4360

> 1.) There appears to be no way to set the flags (inlineSourceMap ,inlineSources ) in visual studio 2013 with the 1.5 TypeScript extension installed. Is this a known bug?

However the discussion moves off to cover debugging issues and I'm not clear whether the first issue is resolved.  Is Visual Studio due to no longer ignore inlineSourceMaps / inlineSources option of `tsconfig.json` when TS 1.8 ships?

The issue is still there with TS 1.7.

**Comment by nycdotnet** [comment]
I know you're trying to use tsconfig.json, but just FYI - all of the options should be available in VS via MSBuild.  https://github.com/Microsoft/TypeScript/wiki/Setting-Compiler-Options-in-MSBuild-projects

**Comment by johnnyreilly** [comment]
Thanks Steve. Happy New Year! It's good that there is another way but in this particular case I have 2 TypeScript projects inside my ASP.Net project. One for tests and one for source and I use different compiler settings for each..

**Comment by johnnyreilly** [comment]
Is this resolved with 1.8 then?

**Comment by mhegazy** [comment]
@johnnyreilly what do you mean resolved? 

as @nycdotnet mentioned all these properties are accessible using through MSBuild using the properties defined in https://github.com/Microsoft/TypeScript/wiki/Setting-Compiler-Options-in-MSBuild-projects, there is just no UI for it.

**Comment by johnnyreilly** [comment]
@mhegazy is this resolved?:

> Is Visual Studio due to no longer ignore inlineSourceMaps / inlineSources option of tsconfig.json when TS 1.8 ships?

**Comment by mhegazy** [comment]
- VS 2013 do not support tsconfig.json. you need to specify all properties as MSBuild properties
- No UI is available in project properties to set these two properties, nether in VS 2013 no VS 2015. [the wiki page](https://github.com/Microsoft/TypeScript/wiki/Setting-Compiler-Options-in-MSBuild-projects) has details on how to do it manually.
- TS 1.8 should add support for debugging JS files generated with inlineSoruceMaps, and inlineSources, which were not supported before.

does that answer your question?

**Comment by johnnyreilly** [comment]
Hi @mhegazy,

Sorry for the delay in responding.  No that doesn't answer my question I'm afraid.  Let me try again:

I'm using Visual Studio 2015 with `tsconfig.json`.  Frustratingly it seems that Visual Studio is **ignoring** options in my `tsconfig.json` - specifically the options related to sourcemaps.  These are the settings in my `tsconfig.json`

```
  "compilerOptions": {
    "inlineSourceMap": true,
    "inlineSources": true,
    "noImplicitAny": false,
    "preserveConstEnums": true,
    "removeComments": false,
    "sourceMap": false
  },
```

If I invoke `tsc` then inline sourcemaps are generated as I would hope.  But if you save a file in Visual Studio then the sourcemaps options are ignored.  In this Visual Studio project I have 2 TypeScript projects; 1 for source code for the site and 1 for testing.  This may be relevant.

You can see this for yourself by taking a look at the PoorClaresArundel project in the following repo: 
https://github.com/johnnyreilly/poorclaresarundel

**Comment by johnnyreilly** [comment]
I can see discussion here that might imply `tsconfig.json` support will work as expected with TS 1.8: https://github.com/Microsoft/TypeScript/issues/5287

Not certain though.  [Support for multiple tsconfig.json files in ASP.NET v5 projects](https://github.com/Microsoft/TypeScript/issues/4714) makes me wonder if ASP.Net pre v5 is not going to get this amend?  Are ASP.Net v5 and ASP.Net pre v5 going to be treated differently in Visual Studio?  That'd be a real shame.

**Comment by paulvanbrenk** [comment]
@johnnyreilly Current plan is to support all project types (at least the ones that ship in Visual Studio. Others will probably work, but if I don't know about it I can't test it ;) )

**Comment by johnnyreilly** [comment]
Great! Is this support due to land in VS 2015 with TypeScript 1.8?

**Comment by mhegazy** [comment]
that is correct.

**Comment by johnnyreilly** [comment]
Boom! Thanks @mhegazy
